tests/devices/test_default_clifford.py

```diff
@@ -94,38 +93,276 @@ def circuit_fn():
+
+    qnode_clfrd = qml.QNode(circuit_fn, dev_c)
+    qnode_qubit = qml.QNode(circuit_fn, dev_q)
+
+    assert np.allclose(qnode_clfrd(), qnode_qubit(), atol=1e-2 if shots else 1e-8)
+
+
+@pytest.mark.parametrize("shots", [1024, 8192, 16384])
+def test_meas_samples(shots):
+    """Test if samples are returned with shots given in the clifford device."""
+
+    @qml.qnode(qml.device("default.clifford", shots=shots))
+    def circuit_fn():
+        qml.BasisState(np.array([1, 1]), wires=range(2))
+        return [
+            qml.sample(wires=[1]),
+            qml.sample(qml.PauliZ(0)),
+            qml.sample(qml.PauliX(0) @ qml.PauliY(1)),
+        ]
+
+    samples = circuit_fn()
+    assert len(samples) == 3
+    assert qml.math.shape(samples[0]) == (shots, 1)
+    assert qml.math.shape(samples[1]) == (shots, 1)
+    assert qml.math.shape(samples[2]) == (shots, 2)
+
+
+@pytest.mark.parametrize("tableau", [True, False])
+@pytest.mark.parametrize("shots", [None, 8192])
+@pytest.mark.parametrize(
+    "ops",
+    [
+        None,
+        qml.PauliY(0),
+        qml.PauliX(0) @ qml.PauliY(1),
+        qml.PauliX(0) @ qml.PauliY(1) @ qml.PauliZ(2),
+    ],
+)
+def test_meas_probs(tableau, shots, ops):
+    """Test if probabilities are returned in the clifford device."""
+
+    dev_c = qml.device("default.clifford", tableau=tableau, shots=shots, seed=24)
+    dev_q = qml.device("default.qubit")
+
+    def circuit_fn():
+        for wire in range(3):
+            qml.PauliX(wire)
+            qml.PauliY(wire)
+            qml.PauliZ(wire)
+        return qml.probs(op=ops) if ops else qml.probs(wires=[0, 1])
+
+    qnode_clfrd = qml.QNode(circuit_fn, dev_c)
+    qnode_qubit = qml.QNode(circuit_fn, dev_q)
+
+    gotten_probs, target_probs = qnode_clfrd(), qnode_qubit()
+    if gotten_probs.shape[-1] != target_probs.shape[-1]:
+        target_probs = qml.math.sum(target_probs.reshape((2,) * len(ops.wires)), axis=-1).reshape(
+            1, -1
+        )
+
+    assert qml.math.allclose(gotten_probs, target_probs, atol=1e-2 if shots else 1e-8)
+
+
+def test_meas_probs_large():
+    """Test if probabilities are returned in the clifford device with target basis states"""
+
+    dev_c = qml.device("default.clifford", seed=24)
+
+    def circuit_fn1():
+        for wire in range(30):
+            qml.PauliX(wire)
+            qml.PauliY(wire)
+            qml.PauliZ(wire)
+        return qml.probs(wires=range(1))
+
+    assert qml.QNode(circuit_fn1, dev_c)().shape == (2,)
+
+    def circuit_fn2(state):
+        for wire in range(30):
+            qml.PauliX(wire)
+            qml.PauliY(wire)
+            qml.PauliZ(wire)
+        return qml.expval(qml.Projector(state, wires=[0, 1]))
+
+    qnode_clfrd = qml.QNode(circuit_fn2, dev_c)
+
+    basis_states = np.array([[0, 0], [1, 0], [0, 1], [1, 1]])
+    probs1 = [qnode_clfrd(state) for state in basis_states]
+    probs2 = [1.0, 0.0, 0.0, 0.0]
+    assert qml.math.allclose(probs1, probs2)
```

Comment:
[non-blocking] This does not seem like a powerful test.

Reply:
Although I think `test_meas_probs` (the test above it) sufficiently tests the probabilities, I have made this one stronger as well.